<a href="https://colab.research.google.com/github/jacquelinepalumbo/Sprint1_MVP/blob/main/Analise_consumidor_Sprint1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sprint 1 - Desenvolvimento do MVP

Título: Análise do Comportamento do Consumidor

O dataset utilizado neste notebook possibilitará uma analise comportamental dos hábitos do consumidor sobre artigos esportivos. 
Nossos dados são estruturados, onde foram extraídos de um banco de dados da plataforma Kaggle.
Trataremos um modelo em aprendizado supervisionado, já que este modelo é construído a partir de um dataset, que são apresentados para um algoritmo na forma de entrada e saída desejada. 

A técnica de estatística descritiva será aplicada para organizar, descrever, analisar e interpretar dados extraídos do dataset, para que conclusões poossam ser feitas.

Utilizaremos Pandas para realizarmos a Análise Exploratória do dataset. NumPy, pacote fundamental, para a computação científica com Python. 
Matplotlib, com as bibliotecas de plotagem gráfica.
O Seaborn, é uma biblioteca de visualização baseada em Matplotlib, para a criação de gráficos atrativos.
Neste notebook, será utilizado a Plotly. É uma biblioteca em Python que permite criar visualizações interativas em diversos formatos. A biblioteca Plotly possui uma interface com uma interação interesante a ser explorada para plots.
Para isso, será gerado a importação destes pacotes e bibliotecas para desenvolver o Notbook.

In [1]:
import numpy as np # Importação do Numpy
import pandas as pd # Importação do Pandas
import matplotlib.pyplot as plt # Importação do Pyplot
import seaborn as sns # Importação do Seaborn
import plotly # Importação do Plotly
import plotly.graph_objs as go # Importação de algumas funcionalidades do Plotly
import plotly.offline as py # Importante plotly libraries do plotly offline

print("Configuração Concluída") # Exibir na tela "Configuração Concluída" 

Configuração Concluída


Agora iremos importar o dataset com que iremos utilizar a partir de uma url para um dataframe. Neste caso, o arquivo foi inserido no repositório do GitHub. O arquivo está em formado csv. O dataset será nomeado como 'customer'.

In [2]:
# url a importar
url_dados = 'https://raw.githubusercontent.com/jacquelinepalumbo/Sprint1_MVP/main/SalesForCourse_quizz_table.csv'

# Carga do dataset através do arquivo csv
customer = pd.read_csv(url_dados)

# Análise Exploratória de Dados

Inicialmente, iremos utilizar alguns comandos da biblioteca Pandas, para exibir algumas estatísticas descritivas e fazer uma rápida análise exploratória do dataset. Os comentários nos blocos de código auxiliam no seu entendimento.

In [3]:
# Verificando que 'customer' é um dataframe
print(type(customer))

# Verificando os tipos de cada coluna do dataframe
customer.dtypes

<class 'pandas.core.frame.DataFrame'>


index                 int64
Date                 object
Year                float64
Month                object
Customer Age        float64
Customer Gender      object
Country              object
State                object
Product Category     object
Sub Category         object
Quantity            float64
Unit Cost           float64
Unit Price          float64
Cost                float64
Revenue             float64
Column1             float64
dtype: object

Nosso dataset é um DataFrame e possui 16 colunas. Nossas variáveis são distribuídas no tipo Float, Inteiro e Objeto. 

Explorando o Dataset.

In [4]:
# Exibindo as 5 primeiras linhas
customer.head()

index      Date    Year     Month  Customer Age Customer Gender  \
0      0  02/19/16  2016.0  February          29.0               F   
1      1  02/20/16  2016.0  February          29.0               F   
2      2  02/27/16  2016.0  February          29.0               F   
3      3  03/12/16  2016.0     March          29.0               F   
4      4  03/12/16  2016.0     March          29.0               F   

         Country       State Product Category     Sub Category  Quantity  \
0  United States  Washington      Accessories  Tires and Tubes       1.0   
1  United States  Washington         Clothing           Gloves       2.0   
2  United States  Washington      Accessories  Tires and Tubes       3.0   
3  United States  Washington      Accessories  Tires and Tubes       2.0   
4  United States  Washington      Accessories  Tires and Tubes       3.0   

   Unit Cost  Unit Price   Cost  Revenue  Column1  
0      80.00  109.000000   80.0    109.0      NaN  
1      24.50   28.500000   49.0     57.0      NaN  
2       3.67    5.000000   11.0     15.0      NaN  
3      87.50  116.500000  175.0    233.0      NaN  
4      35.00   41.666667  105.0    125.0      NaN

In [5]:
# Exibindo as últimas 5 linhas
customer.tail()

index      Date    Year     Month  Customer Age Customer Gender  \
34862  34862  02/07/16  2016.0  February          38.0               M   
34863  34863  03/13/15  2015.0     March          38.0               M   
34864  34864  04/05/15  2015.0     April          38.0               M   
34865  34865  08/30/15  2015.0    August          38.0               M   
34866  34866       NaN     NaN       NaN           NaN             NaN   

      Country           State Product Category    Sub Category  Quantity  \
34862  France  Hauts de Seine            Bikes  Mountain Bikes       2.0   
34863  France  Hauts de Seine            Bikes  Mountain Bikes       1.0   
34864  France  Hauts de Seine            Bikes  Mountain Bikes       3.0   
34865  France  Hauts de Seine            Bikes  Mountain Bikes       1.0   
34866     NaN             NaN              NaN             NaN       NaN   

       Unit Cost   Unit Price    Cost      Revenue  Column1  
34862     1160.0   985.500000  2320.0  1971.000000      NaN  
34863     2049.0  1583.000000  2049.0  1583.000000      NaN  
34864      683.0   560.666667  2049.0  1682.000000      NaN  
34865     2320.0  1568.000000  2320.0  1568.000000      NaN  
34866        NaN          NaN     NaN   641.532095      NaN

In [6]:
# Exibindo as dimensões do dataset
customer.shape

(34867, 16)

In [7]:
# Exibindo os nomes das colunas
customer.columns

Index(['index', 'Date', 'Year', 'Month', 'Customer Age', 'Customer Gender',
       'Country', 'State', 'Product Category', 'Sub Category', 'Quantity',
       'Unit Cost', 'Unit Price', 'Cost', 'Revenue', 'Column1'],
      dtype='object')

In [8]:
# Exibindo a quantidade de linhas por coluna
customer.count()

index               34867
Date                34866
Year                34866
Month               34866
Customer Age        34866
Customer Gender     34866
Country             34866
State               34866
Product Category    34866
Sub Category        34866
Quantity            34866
Unit Cost           34866
Unit Price          34866
Cost                34866
Revenue             34867
Column1              2574
dtype: int64

In [9]:
# Ordenando o dataframe por uma coluna. No caso, a coluna 'Date' foi ordenada.
customer= customer.sort_values(by='Date')

In [10]:
# Ordenando por uma coluna de ordem ascendente (do menor ao maior). No caso, a coluna 'Country' foi ordenada
customer.sort_values(by='Country', ascending=True)

index      Date    Year     Month  Customer Age Customer Gender  \
14576  14576  02/07/16  2016.0  February          38.0               F   
24089  24089  08/06/15  2015.0    August          35.0               M   
23860  23860  08/06/15  2015.0    August          29.0               F   
33495  33495  05/11/16  2016.0       May          25.0               M   
14776  14776  05/26/16  2016.0       May          32.0               F   
...      ...       ...     ...       ...           ...             ...   
9885    9885  05/27/16  2016.0       May          31.0               M   
1406    1406  05/27/16  2016.0       May          32.0               F   
9679    9679  05/27/16  2016.0       May          36.0               F   
21524  21524  12/31/15  2015.0  December          53.0               M   
34866  34866       NaN     NaN       NaN           NaN             NaN   

             Country           State Product Category       Sub Category  \
14576         France         Essonne      Accessories    Tires and Tubes   
24089         France      Val d'Oise         Clothing            Jerseys   
23860         France   Seine (Paris)      Accessories            Helmets   
33495         France  Seine et Marne      Accessories    Tires and Tubes   
14776         France    Val de Marne      Accessories  Bottles and Cages   
...              ...             ...              ...                ...   
9885   United States      Washington      Accessories    Tires and Tubes   
1406   United States      Washington            Bikes     Mountain Bikes   
9679   United States      California            Bikes         Road Bikes   
21524  United States          Oregon      Accessories  Bottles and Cages   
34866            NaN             NaN              NaN                NaN   

       Quantity  Unit Cost   Unit Price    Cost      Revenue  Column1  
14576       2.0      75.00   123.500000   150.0   247.000000      NaN  
24089       1.0    1134.00  1238.000000  1134.0  1238.000000      NaN  
23860       1.0     105.00   150.000000   105.0   150.000000      NaN  
33495       1.0     225.00   265.000000   225.0   265.000000      NaN  
14776       3.0       1.67     2.666667     5.0     8.000000      NaN  
...         ...        ...          ...     ...          ...      ...  
9885        1.0     480.00   643.000000   480.0   643.000000      NaN  
1406        1.0    2295.00  2681.000000  2295.0  2681.000000   1928.0  
9679        1.0    1120.00  1142.000000  1120.0  1142.000000      NaN  
21524       3.0      26.67    30.000000    80.0    90.000000      NaN  
34866       NaN        NaN          NaN     NaN   641.532095      NaN  

[34867 rows x 16 columns]

Nosso Dataset agrega uma quantidade de atributos consideráveis, como data, mês e ano da compra; idade, sexo, país e estado do cliente; categoria do produto, subcategoria do produto, quantidade comprada, custo por unidade, preço de venda por unidade, total do custo comprado pelo cliente e receita gerada da venda.
Podemos trabalhar com estes dados, dividindo-os em tempo relativo de vendas. Países que mais compraram os protudos, receita gerada por protudo e sazonalidade de vendas, por exemplo.

Customer possui 16 colunas e 34867 linhas.

In [11]:
# Exibindo informações sobre o Dataset
print(customer.info()) 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 34867 entries, 2150 to 34866
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   index             34867 non-null  int64  
 1   Date              34866 non-null  object 
 2   Year              34866 non-null  float64
 3   Month             34866 non-null  object 
 4   Customer Age      34866 non-null  float64
 5   Customer Gender   34866 non-null  object 
 6   Country           34866 non-null  object 
 7   State             34866 non-null  object 
 8   Product Category  34866 non-null  object 
 9   Sub Category      34866 non-null  object 
 10  Quantity          34866 non-null  float64
 11  Unit Cost         34866 non-null  float64
 12  Unit Price        34866 non-null  float64
 13  Cost              34866 non-null  float64
 14  Revenue           34867 non-null  float64
 15  Column1           2574 non-null   float64
dtypes: float64(8), int64(1), object(7)
me

In [12]:
# Faz um resumo estatístico do dataset (média, desvio padrão, mínimo, máximo e os quartis)
customer.describe()

index          Year  Customer Age      Quantity     Unit Cost  \
count  34867.000000  34866.000000  34866.000000  34866.000000  34866.000000   
mean   17433.000000   2015.569237     36.382895      2.002524    349.880567   
std    10065.380254      0.495190     11.112902      0.813936    490.015846   
min        0.000000   2015.000000     17.000000      1.000000      0.670000   
25%     8716.500000   2015.000000     28.000000      1.000000     45.000000   
50%    17433.000000   2016.000000     35.000000      2.000000    150.000000   
75%    26149.500000   2016.000000     44.000000      3.000000    455.000000   
max    34866.000000   2016.000000     87.000000      3.000000   3240.000000   

         Unit Price          Cost       Revenue      Column1  
count  34866.000000  34866.000000  34867.000000  2574.000000  
mean     389.232485    576.004532    640.870093   688.054913  
std      525.319091    690.500395    736.640033   774.200897  
min        0.666667      2.000000      2.000000     2.000000  
25%       53.666667     85.000000    102.000000   104.250000  
50%      179.000000    261.000000    319.000000   390.500000  
75%      521.000000    769.000000    902.000000   975.750000  
max     5082.000000   3600.000000   5082.000000  3681.000000

No resumo estatístico do dataset, percebemos que os valores não estão negativos, sendo uma caracterítica positiva e coerente das variáveis presentes.

Nesta etapa, iremos utilizar as técnicas de Pré-Processamento de Dados pelo Pandas, para tratar nossos dados e identificar possíveis probelmas com o nosso dataset, como valores incosistentes e valores faltantes, já observados em algumas linhas e colunas do nosso dataset, representados por "NaN". Note, que ao informar a função abaixo, encontramos diversos valores "null". 
Iremos realizar a trativa para remover valores null.

In [13]:
# Verificando nulls no dataset
customer.isnull().sum()

index                   0
Date                    1
Year                    1
Month                   1
Customer Age            1
Customer Gender         1
Country                 1
State                   1
Product Category        1
Sub Category            1
Quantity                1
Unit Cost               1
Unit Price              1
Cost                    1
Revenue                 0
Column1             32293
dtype: int64

Nosso dataset possui em cada coluna 1 valor null. A coluna representada como 'Column1' possui 32293 valores null. Estes valores missing na coluna em questão trazem relevância ao contexto, em relação as outras colunas? 
'Column1' neste caso, apresenta dados missing que não interferem na análise proposta. Por este motivo, iremos remover os valores null e a 'Column1' por ser uma coluna que não contém nenhum dado que agraga as análises desejadas.

In [14]:
# Eliminando linhas que tenham  algum valores NULL
customer = customer.dropna(how='any')

In [15]:
# Exibindo as 5 primeiras linhas
customer.head()

index      Date    Year    Month  Customer Age Customer Gender  \
2150   2150  01/01/15  2015.0  January          19.0               F   
2055   2055  01/01/16  2016.0  January          39.0               F   
1068   1068  01/01/16  2016.0  January          30.0               F   
1132   1132  01/01/16  2016.0  January          31.0               M   
2604   2604  01/01/16  2016.0  January          33.0               M   

             Country       State Product Category Sub Category  Quantity  \
2150   United States  Washington            Bikes   Road Bikes       1.0   
2055   United States  California      Accessories      Helmets       1.0   
1068  United Kingdom     England         Clothing      Jerseys       1.0   
1132  United Kingdom     England      Accessories      Helmets       2.0   
2604   United States  California      Accessories     Cleaners       3.0   

      Unit Cost  Unit Price    Cost  Revenue  Column1  
2150    1000.00  938.000000  1000.0    938.0    214.0  
2055     210.00  271.000000   210.0    271.0    517.0  
1068     216.00  243.000000   216.0    243.0     86.0  
1132     297.50  379.500000   595.0    759.0    595.0  
2604      13.33   16.333333    40.0     49.0    632.0

In [16]:
# exibindo as 5 últimas linhas
customer.tail() 

index      Date    Year     Month  Customer Age Customer Gender  \
1808   1808  12/31/15  2015.0  December          28.0               M   
1807   1807  12/31/15  2015.0  December          28.0               M   
1806   1806  12/31/15  2015.0  December          28.0               M   
1805   1805  12/31/15  2015.0  December          28.0               M   
1222   1222  12/31/15  2015.0  December          23.0               M   

            Country       State Product Category     Sub Category  Quantity  \
1808  United States  California      Accessories  Tires and Tubes       2.0   
1807  United States  California      Accessories          Helmets       2.0   
1806  United States  California      Accessories  Tires and Tubes       1.0   
1805  United States  California      Accessories  Tires and Tubes       3.0   
1222         France      Loiret            Bikes   Mountain Bikes       3.0   

      Unit Cost  Unit Price    Cost  Revenue  Column1  
1808     455.00  497.000000   910.0    994.0   1138.0  
1807      70.00   75.500000   140.0    151.0   2378.0  
1806      90.00   89.000000    90.0     89.0   1405.0  
1805      81.67   91.333333   245.0    274.0    223.0  
1222     765.00  940.333333  2295.0   2821.0    580.0

In [17]:
# Exibindo informações sobre o Dataset
print(customer.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2574 entries, 2150 to 1222
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   index             2574 non-null   int64  
 1   Date              2574 non-null   object 
 2   Year              2574 non-null   float64
 3   Month             2574 non-null   object 
 4   Customer Age      2574 non-null   float64
 5   Customer Gender   2574 non-null   object 
 6   Country           2574 non-null   object 
 7   State             2574 non-null   object 
 8   Product Category  2574 non-null   object 
 9   Sub Category      2574 non-null   object 
 10  Quantity          2574 non-null   float64
 11  Unit Cost         2574 non-null   float64
 12  Unit Price        2574 non-null   float64
 13  Cost              2574 non-null   float64
 14  Revenue           2574 non-null   float64
 15  Column1           2574 non-null   float64
dtypes: float64(8), int64(1), object(7)
memo

Após realizar o tratramento dos valores missing, o Dataframe tem agora 2574 linhas, em que anteriormente haviam 34867 linhas.

In [18]:
# Exibindo as dimensões do dataset
customer.shape

(2574, 16)

In [19]:
# Verificando nulls no dataset novamente
customer.isnull().sum()

index               0
Date                0
Year                0
Month               0
Customer Age        0
Customer Gender     0
Country             0
State               0
Product Category    0
Sub Category        0
Quantity            0
Unit Cost           0
Unit Price          0
Cost                0
Revenue             0
Column1             0
dtype: int64

In [20]:
# Renomeando os dados do dataframe
customer = customer.rename(columns={'Date':'Data', 'Year':'Ano', 'Month':'Mês', 'Customer Age':'Idade Cliente', 'Customer Gender':'Sexo',
       'Country':'País', 'State':'Estado', 'Product Category':'Categoria Produto', 'Sub Category':'Sub Categoria', 'Quantity':'Quantidade',
       'Unit Cost':'Custo Unitário', 'Unit Price':'Preço Unitário', 'Cost':'Custo', 'Revenue':'Receita'})

In [21]:
# Analisar quais são valores que contém em coluna específicas
customer['Categoria Produto'].value_counts()

Accessories    1653
Bikes           528
Clothing        393
Name: Categoria Produto, dtype: int64

In [22]:
# Analisar quais são valores que contém em coluna específicas
customer['Sub Categoria'].value_counts()

Tires and Tubes      895
Helmets              314
Mountain Bikes       305
Bottles and Cages    241
Jerseys              217
Road Bikes           126
Caps                 113
Touring Bikes         97
Cleaners              88
Fenders               64
Gloves                35
Hydration Packs       25
Vests                 17
Bike Stands           14
Bike Racks            12
Socks                 11
Name: Sub Categoria, dtype: int64

In [23]:
# Analisar quais são valores que contém em coluna específicas
customer['País'].value_counts()

United States     1549
France             430
United Kingdom     344
Germany            251
Name: País, dtype: int64

In [24]:
# Analisar quais são valores que contém em coluna específicas
customer['Estado'].value_counts()

California             860
Washington             513
England                344
Oregon                 164
Hessen                  90
Seine Saint Denis       76
Seine (Paris)           65
Nordrhein-Westfalen     61
Val d'Oise              58
Nord                    50
Yveline                 46
Hauts de Seine          36
Hamburg                 34
Bayern                  33
Saarland                31
Essonne                 24
Seine et Marne          21
Charente-Maritime       14
Loiret                  12
Moselle                 11
Loir et Cher             9
Pas de Calais            5
Ohio                     5
Minnesota                3
Garonne (Haute)          2
Alabama                  2
Wyoming                  2
Brandenburg              2
Val de Marne             1
Name: Estado, dtype: int64

In [25]:
# Analisar quais são valores que contém em coluna específicas
customer['Idade Cliente'].value_counts()

39.0    167
38.0    154
34.0    151
32.0    149
40.0    128
28.0    124
30.0    122
53.0    107
24.0    102
27.0    101
36.0     96
35.0     87
33.0     85
43.0     81
42.0     72
37.0     65
26.0     58
22.0     56
45.0     56
46.0     52
44.0     52
41.0     52
31.0     36
52.0     36
50.0     31
64.0     31
59.0     30
29.0     28
58.0     26
20.0     26
54.0     22
19.0     21
61.0     21
23.0     20
55.0     14
47.0     13
63.0     13
57.0     11
60.0     10
49.0     10
18.0      9
56.0      8
62.0      6
51.0      6
17.0      5
48.0      5
65.0      5
21.0      4
66.0      4
67.0      2
25.0      2
71.0      2
Name: Idade Cliente, dtype: int64

In [26]:
# Analisar quais são valores que contém em coluna específicas
customer['Sexo'].value_counts()

M    1324
F    1250
Name: Sexo, dtype: int64

In [27]:
# Analisar quais são valores que contém em coluna específicas
customer['Data'].value_counts()

01/20/16    19
06/27/16    18
09/13/15    17
06/05/16    17
10/28/15    16
            ..
05/09/15     1
05/08/15     1
05/06/15     1
05/05/15     1
06/18/15     1
Name: Data, Length: 445, dtype: int64

In [28]:
# Exluir uma coluna
customer.drop('Column1', axis = 1, inplace = True) # Coluna 'Column' excluída

In [29]:
# Exluir uma coluna
customer.drop('index', axis = 1, inplace = True) # Coluna 'index' excluída

In [30]:
# Exibindo as 5 primeiras linhas
customer.head()

Data     Ano      Mês  Idade Cliente Sexo            País  \
2150  01/01/15  2015.0  January           19.0    F   United States   
2055  01/01/16  2016.0  January           39.0    F   United States   
1068  01/01/16  2016.0  January           30.0    F  United Kingdom   
1132  01/01/16  2016.0  January           31.0    M  United Kingdom   
2604  01/01/16  2016.0  January           33.0    M   United States   

          Estado Categoria Produto Sub Categoria  Quantidade  Custo Unitário  \
2150  Washington             Bikes    Road Bikes         1.0         1000.00   
2055  California       Accessories       Helmets         1.0          210.00   
1068     England          Clothing       Jerseys         1.0          216.00   
1132     England       Accessories       Helmets         2.0          297.50   
2604  California       Accessories      Cleaners         3.0           13.33   

      Preço Unitário   Custo  Receita  
2150      938.000000  1000.0    938.0  
2055      271.000000   210.0    271.0  
1068      243.000000   216.0    243.0  
1132      379.500000   595.0    759.0  
2604       16.333333    40.0     49.0

As variáveis da coluna 'Data' são o tipo object e iremos transformá-la em str para trabalhar com esses dados. Um objeto da classe date representa uma data(ano, mês e dia). Neste caso, iremos trabalhar em mês/ano, para visualizar as vendas mensais. A análise será feita com dados mensais. Claro que os dias dos meses são importantes para, por exemplo, verificar os dias da semana em que há o maior faturamento. Porém, para uma visualização mais generalista, trabalharemos com a visualização mês/ano.


In [31]:
# Transformar a coluna Data(formato inteiro) para o formato datetime
customer.Data = pd.to_datetime(customer.Data)

In [32]:
# Visualização da coluna Data para o formato datetime
customer.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2574 entries, 2150 to 1222
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Data               2574 non-null   datetime64[ns]
 1   Ano                2574 non-null   float64       
 2   Mês                2574 non-null   object        
 3   Idade Cliente      2574 non-null   float64       
 4   Sexo               2574 non-null   object        
 5   País               2574 non-null   object        
 6   Estado             2574 non-null   object        
 7   Categoria Produto  2574 non-null   object        
 8   Sub Categoria      2574 non-null   object        
 9   Quantidade         2574 non-null   float64       
 10  Custo Unitário     2574 non-null   float64       
 11  Preço Unitário     2574 non-null   float64       
 12  Custo              2574 non-null   float64       
 13  Receita            2574 non-null   float64       
dtypes: da

In [33]:
# Usar a função to_period para trabalhar com o dataframe em mês/ano
customer.Data.dt.to_period('M')

2150    2015-01
2055    2016-01
1068    2016-01
1132    2016-01
2604    2016-01
         ...   
1808    2015-12
1807    2015-12
1806    2015-12
1805    2015-12
1222    2015-12
Name: Data, Length: 2574, dtype: period[M]

In [34]:
# Criando uma nova coluna ['cliente_data'] do tipo str para trabalhar no dataframe
customer['cliente_data'] = customer.Data.dt.to_period('M').astype(str)

In [35]:
# Criar a variável vendas_por_mes: que é a soma do valor dos produtos agrupados por mês pela receita ['Receita']
vendas_por_mes = customer.groupby(by='cliente_data').Receita.sum()

In [36]:
# Visualizando os primeiros valores de vendas_por_mes
vendas_por_mes.head()

cliente_data
2015-01    13547.0
2015-02    21858.0
2015-03    12254.0
2015-04    19440.0
2015-05    26033.0
Name: Receita, dtype: float64

In [37]:
# Exibindo os valores somados em vendas_por_mes
vendas_por_mes.values

array([ 13547.,  21858.,  12254.,  19440.,  26033.,  52706.,  96932.,
       150808., 121612., 120589., 122323., 183515., 176736., 112815.,
       107259., 132992., 142105., 161668.,  36195.])

In [38]:
# Exibindo os meses que foram somados
vendas_por_mes.index.item

<bound method IndexOpsMixin.item of Index(['2015-01', '2015-02', '2015-03', '2015-04', '2015-05', '2015-06',
       '2015-07', '2015-08', '2015-09', '2015-10', '2015-11', '2015-12',
       '2016-01', '2016-02', '2016-03', '2016-04', '2016-05', '2016-06',
       '2016-07'],
      dtype='object', name='cliente_data')>

In [39]:
# Exibindo as 5 primeiras linhas do dataset com a nova coluna ['cliente_data']
customer.head() 

Data     Ano      Mês  Idade Cliente Sexo            País  \
2150 2015-01-01  2015.0  January           19.0    F   United States   
2055 2016-01-01  2016.0  January           39.0    F   United States   
1068 2016-01-01  2016.0  January           30.0    F  United Kingdom   
1132 2016-01-01  2016.0  January           31.0    M  United Kingdom   
2604 2016-01-01  2016.0  January           33.0    M   United States   

          Estado Categoria Produto Sub Categoria  Quantidade  Custo Unitário  \
2150  Washington             Bikes    Road Bikes         1.0         1000.00   
2055  California       Accessories       Helmets         1.0          210.00   
1068     England          Clothing       Jerseys         1.0          216.00   
1132     England       Accessories       Helmets         2.0          297.50   
2604  California       Accessories      Cleaners         3.0           13.33   

      Preço Unitário   Custo  Receita cliente_data  
2150      938.000000  1000.0    938.0      2015-01  
2055      271.000000   210.0    271.0      2016-01  
1068      243.000000   216.0    243.0      2016-01  
1132      379.500000   595.0    759.0      2016-01  
2604       16.333333    40.0     49.0      2016-01

**CONSTRUÇÃO DOS GRÁFICOS - ANÁLISE**

 Os gráficos gerados serão utilizados na biblioteca Pyplot.

# Gráfico de Barras

O gráfico de colunas permite contar o número de objetos (ou observações) em dada categoria, neste caso, o valor da receita gerada, em meses, pelos anos de 2015 e 2016.

Gráfico de Barras referente a receita gerada em 2015 e 2016.

In [123]:
# Atribuindo os valores de x e y
data = [go.Bar(x=vendas_por_mes.index,
               y=vendas_por_mes.values,
               marker = {'color': 'DeepSkyBlue'} # definindo a cor           
              )
       ]

# Incluindo Título
receita_fig_layout = go.Layout(title='Receita Mensal 2015-2016 (em k $)',
                               titlefont=dict(size=30), # tamanho do título
                               autosize=True, # auto-ajuste das dimensões do grafico
                               yaxis={'title':'Receita USD'}, # título ao eixo y
                               xaxis={'title':'Período'} # título ao eixo x
                              )

# Definindo objeto figura para print
fig1 = go.Figure(data=data, layout=receita_fig_layout)

# print gráfico
py.iplot(fig1)


O primeiro gráfico gerado nos mostra uma tendência de receita gerada conforme os meses avançam. A bibloteca Pyplot nos permite interagir com o gráfico, ao passar o mouse, podemos analisar os valores da receita mensal gerada.

Destacando barras com o valor abaixo da média de vendas

In [124]:
# Definindo 'cores' para destaque de barras. [] representa uma definição vazia de cores
cores = []

In [125]:
# Média de Vendas 2015-2016
media = vendas_por_mes.values.mean() # mean: faz a média dos valores

print (media)

95336.15789473684


Para definir cores em destaque, utilizaremos a Condicional If/Else. Iremos propor uma condição: caso os meses que não atinjam o valor médio, será marcado pela cor vermelha. Os demais, serão mantidos com a cor atual.
O método append() acrescenta um elemento ao final da lista (neste caso, a cor).

In [126]:
# Parâmetro valores vendas_por_mês
for x in vendas_por_mes.values: 
  if x < media:  # Se x for menor que a média
    cores.append('red') # 'Cores' será destacado em vermelho 

  else: # Senão
    cores.append('DeepSkyBlue')  #'Cores' será permanecerá a cor definida anteriormente

In [127]:
# Exisbindo novas cores a serem plotadas após a função If/Else
cores

['red',
 'red',
 'red',
 'red',
 'red',
 'red',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'DeepSkyBlue',
 'red']

Gráfico de Barras plotado com destaque de barras

In [128]:
# Atribuindo os valores de x e y
data = [go.Bar(x=vendas_por_mes.index,
               y=vendas_por_mes.values,
               marker = {'color': cores} # definindo cores com novos parâmetros
                        
              )
       ]

# Incluindo Título
receita_fig_layout = go.Layout(title='Receita Mensal 2015-2016 (em k $)',
                               titlefont=dict(size=30), # tamanho do título
                               autosize=True, # auto-ajuste das dimensões do grafico
                               yaxis={'title':'Receita'}, # título do eixo y
                               xaxis={'title':'Período'} # título do eixo x
                              )

# Definindo objeto figura para print
fig1 = go.Figure(data=data, layout=receita_fig_layout)


py.iplot(fig1)

Note, que no primeiro semestre de 2015, a receita gerada foi abaixo da média. Já no segundo semestre foi o oposto. Os meses atingiram e ultrapassaram a média de $ 95k, mantendo uma tendência por 11 meses. 
Após esse período, o último dado gerado (Julho/2016) apresenta uma queda brusca da receita gerada. O que causa curiosidade a descobrir o que levou essa queda brusca e repentina ? Falência? Mudança de hábidos do consumidor? Concorrência? Alta nos produtos?

Destacando o maior e o menor valor da receita em 2015-2016. A função max() retorna o item com o valor mais alto. A função min() retorna o item com o valor mais baixo.

In [129]:
# Calculando o valor máximo das vendas_por_mes, representada pela variável maximo_de_vendas
maximo_de_vendas = vendas_por_mes.values.max()

# Exibindo o valor
print(maximo_de_vendas)

183515.0


A função zip() retorna um objeto zip, que é um iterador de tuplas onde o primeiro item em cada iterador passado é emparelhado e, em seguida, o segundo item em cada iterador passado é emparelhado. No caso x = vendas_por_mes.values == maximo_de_vendas e y = vendas_por_mes.index. == mes_maximo_de_vendas (será descatado).

In [130]:
# Definindo novas cores(reset)
cores = [] 

# Parâmetro valores vendas por mês e receita
for x,y in zip(vendas_por_mes.values, vendas_por_mes.index): 
  if x == maximo_de_vendas:  # Se x for igual a máxima de vendas
       mes_maximo_de_vendas = y   # o mes máximo de vendas será igual a y
       cores.append('RoyalBlue') # 'Cores' será destacado em azul

  else: # Senão
    cores.append('LightGrey')  # os demais valores serão destacados em cinza

In [131]:
# Exibindo o mês com a maior receita de vendas
mes_maximo_de_vendas 

'2015-12'

In [132]:
# Exibindo a lista [cores] que serão plotadas
cores 

['LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'RoyalBlue',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey']

Gráfico de Barras plotado com destaque de barras

In [133]:
# Atribuindo os valores de x e y
max_vendas = [go.Bar(x=vendas_por_mes.index,
                     y=vendas_por_mes.values,
                     marker = {'color': cores, # definindo cores com novos parâmetros
                         'line': {'color': '#333', # definindo cor da borda
                         'width': 1,}   # definindo espessura da borda 
                              }                       
                    )
             ]

# Incluindo Título
max_fig_layout = go.Layout(title='Receita Mensal 2015-2016 (em k $)', # título do gráfico
                           titlefont=dict(size=30), # tamanho da fonte do título
                           autosize=True, # auto-ajuste das dimensões do grafico
                           yaxis={'title':'Receita'}, # título  eixo y
                           xaxis={'title':'Período'}, # título  eixo x
                           annotations = [{'text': 'Mês destaque de vendas', # função que gera uma anotação no gráfico
                                           'x': mes_maximo_de_vendas, # definindo a posição da anotação eixo x
                                           'y': maximo_de_vendas} # definindo a posição da anotação eixo y
                                         ] 
                          )

# Definindo objeto figura para print 
fig1 = go.Figure(data=max_vendas, layout=max_fig_layout)

py.iplot(fig1)

O mês campeão de vendas foi dezembro/2015 e foi destacada como mostrado na figura anterior.

In [134]:
# Calculando o valor mínimo das vendas_por_mes, representada pela variável minimo_de_vendas
minimo_de_vendas = vendas_por_mes.values.min()

# Exibindo o valor
print(minimo_de_vendas)

12254.0


In [135]:
# Definindo novas cores(reset)
cores = [] 

# Parâmetro valores vendas por mês e receita
for x,y in zip(vendas_por_mes.values, vendas_por_mes.index): 
  if x == minimo_de_vendas:  # Se x for igual ao mínimo de vendas
       mes_minimo_de_vendas = y   # o mes mínimo de vendas será igual a y
       cores.append('FireBrick') # 'Cores' será destacado em vermelho

  else: # Senão
    cores.append('LightGrey')  # os demais valores serão destacados em cinza

In [136]:
# Exibindo a lista [cores] que serão plotadas
cores

['LightGrey',
 'LightGrey',
 'FireBrick',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey',
 'LightGrey']

In [137]:
# Atribuindo os valores de x e y
min_vendas = [go.Bar(x=vendas_por_mes.index,
                     y=vendas_por_mes.values,
                     marker = {'color': cores, # definindo cores com novos parâmetros
                         'line': {'color': '#333', # definindo cor da borda
                         'width': 1,}   # definindo espessura da borda 
                              }                       
                    )
             ]

# Incluindo Título
min_fig_layout = go.Layout(title='Receita Mensal 2015-2016 (em k $)', # título do gráfico
                           titlefont=dict(size=30), # tamanho da fonte do título
                           autosize=True, # auto-ajuste das dimensões do grafico
                           yaxis={'title':'Receita'}, # título  eixo y
                           xaxis={'title':'Período'}, # título  eixo x
                           annotations = [{'text': 'Mês pior de vendas', # função que gera uma anotação no gráfico
                                           'x': mes_minimo_de_vendas, # definindo a posição da anotação eixo x
                                           'y': minimo_de_vendas} # definindo a posição da anotação eixo y
                                         ] 
                          )

# Definindo objeto figura para print 
fig1 = go.Figure(data=min_vendas, layout=min_fig_layout)

py.iplot(fig1)

O mês pior de vendas foi março/2015 e foi destacada como mostrado na figura anterior.

# Histograma 

Histograma é uma espécie de gráfico de barras que demonstra uma distribuição de frequências. No histograma, a base de cada uma das barras representa uma classe, no caso a Idade dos consumidores e a altura representa a frequência absoluta com que o valor de cada classe ocorre.

In [112]:
# Histograma usando Pyplot Librarie
# Plotando o histograma Idade dos Clientes
data = [go.Histogram(x=customer['Idade Cliente'])]

# Incluindo título do gráfico e legenda
layout = go.Layout(title='Idade Dos Cosumidores', # Título do gráfico
                   titlefont=dict(size=30), # tamanho do título
                   yaxis={'title':'Frequência'}, # título do eixo y
                   xaxis={'title':'Idade Clientes'}, # título do eixo x
                  ) 

# Definindo objeto figura para print 
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

# Gráfico de Pizza

Mostraremos a visualização de um valor relativo de cada categoria estabelecida em relação a um todo. Neste caso, a proporção de clientes que consomem os produtos por sexo.

In [113]:
# Analisar quais são valores que contém em coluna específicas
sexo_customer = customer['Sexo'].value_counts()

print(sexo_customer)

M    1324
F    1250
Name: Sexo, dtype: int64


In [114]:
# Definindo cores para o gráfico
cores = ['#B0C4DE', '#DDA0DD'] 

# Construção do Gráfico de Pizza
trace = go.Pie(labels = sexo_customer.index, # labels/rótulos (Feminino/Masculino)
               values = sexo_customer.values, # valores
               marker = {'colors': cores, # inserindo as cores
                         'line': {'color': '#000000', 'width': 1} # definindo cor da borda e espessura da borda
                        }
              )
# Armazenando gráfico em uma lista
data = [trace]

# Layout
layout = go.Layout(title='Cosumidores por Sexo', # título do gráfico
                   titlefont=dict(size=30), # tamanho do título
                   autosize=True # auto-ajuste das dimensões do gráfico
                   )

# Definindo objeto figura para print 
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

Os consumidores são quase proporcionalmente muito semelhantes em questão ao sexo

# Gráfico de Barras

Distinção de países geradores da receita

In [115]:
# Criar a variável vendas_por_paises: que é a soma da receita gerada ['Receita'] pelos países ['País']
vendas_por_paises = customer.groupby(by='País').Receita.sum()/1e3

In [116]:
# Exibindo os valores
vendas_por_paises.head()

País
France            391.566
Germany           245.817
United Kingdom    298.926
United States     875.078
Name: Receita, dtype: float64

In [117]:
# Criando gráfico

# Atribuindo os valores de x e y
data = [go.Bar(x = vendas_por_paises.index,
               y = vendas_por_paises.values,
               marker = {'color': '#BA55D3', # definindo cor das barras
                         'line': {'color': '#333', # definindo cor da borda
                         'width': 1} # definindo espessura da borda
                         }   
              )
      ]
# Criando Layout
layout = go.Layout(title='Vendas Países em Mil USD (2015-2016)', # título do gráfico
                   yaxis={'title':'USD', # título do eixo y
                          'tickprefix':'$', # adicionar nos valores do eixo y o "$" no início
                          'ticksuffix':'k'} # adicionar nos valores do eixo y o "$" no final
) 
# Definindo objeto figura para print 
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

O maior gerador de receita é proveniente dos cosumidores que estão nos Estados Unidos. Será que é devido ao alto consumo desdes produtos? Grande quantidade de lojas? 

# Gráfico Scatter 

O gráfico de dispersão é útil para verificarmos se existe alguma relação entre duas variáveis numéricas. Neste caso, utilizaremos para distinção de estados geradores da receita

Gráfico para análise da distribuição de Estados por países

In [121]:
# Criando gráfico

# Atribuindo os valores de x e y
trace =  go.Scatter(x = customer['Estado'],
                    y = customer['País'],
                    mode = 'markers', # apresentação em plots/markers
                    marker = {'color': '#9400D3'} # definindo uma cor
                   )

# Armazenando o gráfico em uma lista
data = [trace]

# Criando Layout
layout = go.Layout(title='Distribução de Clientes por Países', # título do gráfico 
                   yaxis={'title':'Países'}, # título do gráfico eixo y
                   xaxis={'title': 'Estados'}) # título do gráfico eixo x

# Definindo objeto figura para print 
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

Há mais estados da FRança que consomem os protudos do que nos Estados e Alemanha.

In [62]:
#Criar a variável vendas_por_estados: que é a soma da receita gerada ['Receita'] pelos Estados ['Estado']
vendas_por_estados = customer.groupby(by='Estado').Receita.sum()/1e3

In [63]:
# Exibindo os valores
vendas_por_estados

Estado
Alabama                  1.214
Bayern                  33.161
Brandenburg              3.162
California             468.387
Charente-Maritime       10.548
England                298.926
Essonne                 17.265
Garonne (Haute)          3.236
Hamburg                 39.288
Hauts de Seine          49.365
Hessen                  86.363
Loir et Cher             5.845
Loiret                  17.620
Minnesota                2.161
Moselle                 11.098
Nord                    52.305
Nordrhein-Westfalen     59.765
Ohio                     5.428
Oregon                 112.327
Pas de Calais            1.279
Saarland                24.078
Seine (Paris)           57.989
Seine Saint Denis       67.821
Seine et Marne          18.413
Val d'Oise              18.137
Val de Marne             0.861
Washington             283.387
Wyoming                  2.174
Yveline                 59.784
Name: Receita, dtype: float64

In [122]:
# Criando gráfico Scatter

# Atribuindo os valores de x e y
trace1 =  go.Scatter(x = vendas_por_estados.index,
                    y = vendas_por_estados.values,
                    mode = 'markers', # apresentação em plots/markers
                    marker = {'color': '#FF4500'} # definindo cor
                   )

# Armazenando o gráfico em uma lista
data = [trace1]

# Criando Layout
layout = go.Layout(title='Distribução de Clientes/Estados pela Receita $ 2015-2016', # título do gráfico
                   yaxis={'title':'Receita USD', # título do gráfico eixo y
                          'ticksuffix':'k'}, # adicionar nos valores do eixo y o "$" no final
                   xaxis={'title': 'Estados'}) # título do gráfico eixo x

# Definindo objeto figura para print 
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

O gráfico scatter mostra uma distribuição de receita heterogênea entre os estados. Notamos que o estado da Califórnia e Washington representam os maiores geradores da receita da empresa nos Estados Unidos. 
Hessen na Alemenha e Seine Saint Denis na França;
No Reino Unido, temos apenas a Inglaterra como estado gerador de renda no País.
Por mais que a França contém mais Estados onde há aquisição dos produtos do que os Estados do Estados Unidos, não demostra que são os maiores geradores de renda. 

# Outras Análises

Criando um dicionário dos Continentes para avaliar a receita por Continente.

In [65]:
# Dicionário para atribuir continentes aos países listados
# Exibindo os valores somados em vendas_por_estado
vendas_por_estados.index

Index(['Alabama', 'Bayern', 'Brandenburg', 'California', 'Charente-Maritime',
       'England', 'Essonne', 'Garonne (Haute)', 'Hamburg', 'Hauts de Seine',
       'Hessen', 'Loir et Cher', 'Loiret', 'Minnesota', 'Moselle', 'Nord',
       'Nordrhein-Westfalen', 'Ohio', 'Oregon', 'Pas de Calais', 'Saarland',
       'Seine (Paris)', 'Seine Saint Denis', 'Seine et Marne', 'Val d'Oise',
       'Val de Marne', 'Washington', 'Wyoming', 'Yveline'],
      dtype='object', name='Estado')

In [66]:
# Dicionário para atribuir estados aos países listados
relContinente = {
    "Alabama": "USA",
    "Bayern": "Europe",
    "Brandenburg": "Europe",
    "California": "USA",
    "Charente-Maritime" : "Europe",  
    "England": "USA",
    "Essonne": "Europe", 
    "Garonne (Haute)": "Europe",   
    "Hamburg": "Europe",
    "Hauts de Seine": "Europe",   
    "Hessen": "Europe",
    "Loir et Cher": "Europe",   
    "Loiret": "Europe",   
    "Minnesota": "USA",
    "Moselle": "Europe",   
    "Nord": "Europe",   
    "Nordrhein-Westfalen": "Europe",
    "Ohio": "USA",
    "Oregon": "USA",
    "Pas de Calais": "Europe",   
    "Saarland": "Europe",
    "Seine (Paris)": "Europe",   
    "Seine Saint Denis": "Europe",   
    "Seine et Marne": "Europe",   
    "Val d'Oise": "Europe",   
    "Val de Marne": "Europe",   
    "Washington": "USA",
    "Wyoming": "USA",
    "Yveline": "Europe"
                    }

In [67]:
# Cria uma coluna associando os continentes aos países
customer["Continente"] = customer.apply(lambda row: relContinente[row["Estado"]], axis = 1) # axis=1 refere-se a coluna

# Examina o dataframe resultante
customer

Data     Ano       Mês  Idade Cliente Sexo            País  \
2150 2015-01-01  2015.0   January           19.0    F   United States   
2055 2016-01-01  2016.0   January           39.0    F   United States   
1068 2016-01-01  2016.0   January           30.0    F  United Kingdom   
1132 2016-01-01  2016.0   January           31.0    M  United Kingdom   
2604 2016-01-01  2016.0   January           33.0    M   United States   
...         ...     ...       ...            ...  ...             ...   
1808 2015-12-31  2015.0  December           28.0    M   United States   
1807 2015-12-31  2015.0  December           28.0    M   United States   
1806 2015-12-31  2015.0  December           28.0    M   United States   
1805 2015-12-31  2015.0  December           28.0    M   United States   
1222 2015-12-31  2015.0  December           23.0    M          France   

          Estado Categoria Produto    Sub Categoria  Quantidade  \
2150  Washington             Bikes       Road Bikes         1.0   
2055  California       Accessories          Helmets         1.0   
1068     England          Clothing          Jerseys         1.0   
1132     England       Accessories          Helmets         2.0   
2604  California       Accessories         Cleaners         3.0   
...          ...               ...              ...         ...   
1808  California       Accessories  Tires and Tubes         2.0   
1807  California       Accessories          Helmets         2.0   
1806  California       Accessories  Tires and Tubes         1.0   
1805  California       Accessories  Tires and Tubes         3.0   
1222      Loiret             Bikes   Mountain Bikes         3.0   

      Custo Unitário  Preço Unitário   Custo  Receita cliente_data Continente  
2150         1000.00      938.000000  1000.0    938.0      2015-01        USA  
2055          210.00      271.000000   210.0    271.0      2016-01        USA  
1068          216.00      243.000000   216.0    243.0      2016-01        USA  
1132          297.50      379.500000   595.0    759.0      2016-01        USA  
2604           13.33       16.333333    40.0     49.0      2016-01        USA  
...              ...             ...     ...      ...          ...        ...  
1808          455.00      497.000000   910.0    994.0      2015-12        USA  
1807           70.00       75.500000   140.0    151.0      2015-12        USA  
1806           90.00       89.000000    90.0     89.0      2015-12        USA  
1805           81.67       91.333333   245.0    274.0      2015-12        USA  
1222          765.00      940.333333  2295.0   2821.0      2015-12     Europe  

[2574 rows x 16 columns]

In [68]:
#Criar a variável vendas_continente: que é a soma da receita gerada ['Receita'] pelos Continentes ['Continente']
vendas_continente = customer.groupby(by='Continente').Receita.sum()/1e3

Grafico de Barras

In [69]:
# Criando gráfico

# Atribuindo os valores de x e y
data = [go.Bar(x = vendas_continente.index,
                y = vendas_continente.values,
                marker = {'color': '#F08080', # definindo cor
                          'line': {'color': '#333', # definindo cor da borda
                          'width': 1}} # definindo espessura da borda
              )
       ]

# Criando Layout
layout = go.Layout(title='Vendas América x Europa USD (2015-2016)', # título do gráfico
                   yaxis={'title':'Receita', # título no eixo y
                          'tickprefix':'$',# adicionar nos valores do eixo y o "$" no início) 
                          'ticksuffix':'k'} # adicionar nos valores do eixo y o "$" no final
                  )
 
# Definindo objeto figura para print
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

Ao somar a receita dos estados da Europa e os estados da USA, os maiores geradores de receita se concentram da USA.

Categoria de produtos mais vendidos

In [70]:
# Criando uma variável vendas_por_categoria com o método count() retorna o número de elementos com o valor especificado. ['Categoria Produto'] por ['Quantidade']
vendas_por_categoria = customer.groupby(by='Categoria Produto').Quantidade.count()

In [71]:
# Exibindo variável
vendas_por_categoria

Categoria Produto
Accessories    1653
Bikes           528
Clothing        393
Name: Quantidade, dtype: int64

In [72]:
# Colocando em sentido horário o maior valor
vendas_por_categoria.sort_values(ascending=False, inplace=True)

**Grafico de Pizza:** análise de proporção das Categorias mais vendidas

In [73]:
# Gráfico de Pizza

# Definindo cores para o gráfico
cores = ['#3CB371', '#FFFACD', '#FF6347'] 

# Atribuindo os valores de x e y
trace2 = go.Pie(labels = vendas_por_categoria.index, # index
               values = vendas_por_categoria.values, # valores
               marker = {'colors': cores, # inserindo as cores
                         'line': {'color': '#000000', 'width': 1} # definindo cor de borda e linha de borda
                        },
               direction='clockwise' # direção dos valores em sentido horário
               )

# Armazenando gráfico em uma lista
data = [trace2]

# Criando Layout
layout = go.Layout(title='Classificação de Categorias Mais Vendidas')

# Definindo objeto figura para print
fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

A categoria Acessórios representa o maior índice de requisição pelos consumidores

**Aálise dos itens que geram a maior receita**

In [74]:
#Criar a variável vendas_sub_categoria: que é a soma da receita gerada ['Receita'] pela Sub Categoria ['Sub Categoria']
vendas_sub_categoria = customer.groupby(by='Sub Categoria').Receita.sum()/1e3

In [75]:
# Colocando os valores em ordem crescente
vendas_sub_categoria.sort_values(ascending=True, inplace=True)

In [76]:
# Mostrando as primeiras linhas
vendas_sub_categoria.head() 

Sub Categoria
Socks           1.657
Cleaners       12.546
Bike Stands    14.436
Gloves         14.574
Vests          18.740
Name: Receita, dtype: float64

In [77]:
# Atribuindo valores de y e x 
data = [go.Bar(y=vendas_sub_categoria.index,
               x=vendas_sub_categoria.values,                      
               orientation='h', # gráfico na horizontal
               marker = {'color': 'PaleVioletRed', # definindo cor
                         'line': {'color': '#333', #definindo cor da borda
                         'width': 1,}   #definindo espessura da borda                               
                        }                      
              )
       ]

# Criando Layout
config_layout = go.Layout(title= 'Receita Produtos $K (2015-2016)', # título do gráfico
                          titlefont=dict(size=30), # tamanho do título       
                          yaxis={'title':'Produtos',}, # título do eixo y    
                          xaxis={'title':'Faturamento $ K', # título do eixo x
                                 'tickprefix':'$', # Adicionando o '$' no eixo x
                                 'ticksuffix':'k'} # adicionar nos valores do eixo x o "$" no final
                          )

# Definindo objeto figura para print
fig= go.Figure(data = data, layout=config_layout)

py.iplot(fig)


Ao análisar os item Acessários como mais requisitados pelos consumidores. Os produtos que geram maior receita se enquadram na categoria Bike, que é item que gera maior receita a empresa = mountain bikes. Seguido por Acessórios, no item Tire and Tubes

# Conclusão

O dataset apresenta uma ampla perspectiva sobre os hábitos dos consumidores. Ao explorar o dataset nomeado como 'customer', nos revelou que é um mercado variado e com opções para atingir diversos tipos de consumidores. Foi explorado que a carteira de clientes é diversificada e não específica, pois abrange diversas faixas etárias de consumidores e referente ao sexo, possui uma igualdade de gênero para aquisição das categorias dos produtos. 
ALém disso, foi abordado a receita gerada pelos produtos consumidos e que mostra-se um mercado em tendência para investimento em diversas áreas do globo. Ao mostrar uma país, como Estados Unidos, que possui elevado consumo, mostrou-se verdadeiro a receita gerada para a Empresa, pois há uma demanda ascendente dos produtos.
Há diversas perspectivas que podem ser exploradas no dataset, como produto com maior desconto, distribuição das categorias por estados, dias da semana que há maior receita gerada, entre outros pontos.
